### DISCUSSION TITLE
Holding Event not working

### DISCUSSION THREAD

**Discussion by Nemonek** [AvaloniaUI/Avalonia#12316] [Q&A]
Hi, I was trying to create an app in which you can create and drag a square around, and I wanted to distinguish between the holding click and the single click, but when I use the holding event, even if I hold the click on the rectangle it doesn't works, it only works if I use the PointerPressed event; am I doing something wrong?
```C#
using Avalonia;
using Avalonia.Controls;
using Avalonia.Controls.Shapes;
using Avalonia.Input;
using Avalonia.Interactivity;
using Avalonia.Media;

namespace AvaloniaUI_Creator
{
    public partial class MainWindow : Window
    {
        private Rectangle[] array;
        private int counter;
        public MainWindow()
        {
            InitializeComponent();
            counter = 0;
            array = new Rectangle[10];

        }

        private Rectangle add()
        {
            Rectangle rec = new()
            {
                Height = 200,
                Width = 200,
                Fill = Brushes.Red
            };

            //rec.AddHandler(Rectangle.PointerPressedEvent, PointerPress);
            rec.AddHandler(Rectangle.HoldingEvent, PointerPress);
            return rec;
        }

        Control drag = null;
        Point offset;

        private void PointerPress(object sender, PointerPressedEventArgs e)
        {
            var x =
            drag = sender as Control;
            offset = e.GetPosition(MainCanvas);
            offset = new(
                offset.Y - Canvas.GetTop(drag),
                offset.X - Canvas.GetLeft(drag));

        }
        private void PointerMoving(object sender, PointerEventArgs e)
        {
            if (drag == null)
                return;
            Point position = e.GetPosition(sender as InputElement);
            Canvas.SetTop(drag, position.Y - offset.Y);
            Canvas.SetLeft(drag, position.X - offset.X);

        }

        private void released(object sender, PointerReleasedEventArgs e)
        {
            drag = null;

        }
        private void NewCube(object sender, RoutedEventArgs e)
        {
            array[counter] = add();
            Canvas.SetTop(array[counter], 0);
            Canvas.SetLeft(array[counter], 0);
            MainCanvas.Children.Add(array[counter]);

            counter++;
        }
    }
}
```

**Comment by nir34** [comment]
Is there any progress?

**Reply by workgroupengineering** [reply]
https://github.com/AvaloniaUI/Avalonia.Labs/blob/main/samples/Avalonia.Labs.Catalog/Views/GestureView.axaml

**Reply by nir34** [reply]
Thanks,Your provided code perfectly solved the problem, but I'm curious about the use of the Holding event?

**Reply by workgroupengineering** [reply]
There is a build in Gesture to use it with which you should enable it with [`Gestures.SetIsHoldingEnabled`](https://github.com/AvaloniaUI/Avalonia/blob/175673fa1dcd13b9c07060319e57231601580202/src/Avalonia.Base/Input/Gestures.cs#L99C28-L99C47)

**Comment by GIL0487** [comment]
Apologies, could you please elaborate the answer? I have a simple AvaloniaApplication and I added a button in my MainWindow
`<Button Margin="50, 50, 0, 0"
            Content="TEST"
            Height="50"
            Width="50"
            Background="Yellow"
            Foreground="Red"
            HorizontalAlignment="Left"
            VerticalAlignment="Top"
            Gestures.IsHoldWithMouseEnabled="True"
            Gestures.IsHoldingEnabled="True"
            Holding="InputElement_OnHolding">
    </Button>`

No matter what I try, the holding event is never generated and no code in the InputElement_OnHolding handler is executed.

Thanks
